backend/src/main/java/com/redhat/cloud/notifications/routers/internal/DuplicateNameMigrationResource.java

```diff
@@ -0,0 +1,147 @@
+package com.redhat.cloud.notifications.routers.internal;
+
+import com.redhat.cloud.notifications.auth.ConsoleIdentityProvider;
+import com.redhat.cloud.notifications.models.EndpointType;
+import com.redhat.cloud.notifications.routers.internal.models.DuplicateNameMigrationReport;
+import io.quarkus.logging.Log;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+@RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)
+@Path(API_INTERNAL + "/duplicate-name-migration")
+public class DuplicateNameMigrationResource {
+
+    final String ack = "i-am-sure-i-want-to-run-the-migration";
+
+    @Inject
+    EntityManager entityManager;
+
+    @PostConstruct
+    public void initialize() {
+        Log.infof("Duplicate name migration resource ack with value [%s]", ack);
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    public DuplicateNameMigrationReport migrateDuplicateNames(@QueryParam("ack") String ack) {
+        if (!Objects.equals(this.ack, ack)) {
+            throw new BadRequestException(String.format("Invalid ack provided - should be [%s]", ack));
+        }
+
+        DuplicateNameMigrationReport migrationReport = new DuplicateNameMigrationReport();
+
+        Log.infof("Starting migration for duplicated names on behavior groups and integrations.");
+
+        while (true) {
```

Comment:
Could you add a log entry inside that loop? Even if that's very unlikely, I'd like to know if we are stuck in an infinite or very long loop.

Reply:
There is this log: 

https://github.com/RedHatInsights/notifications-backend/pull/1528/files#diff-5cf2fbcd4bdeaedc0cc1387624379dbfdbeeb9b2c97b98dd5ca0ae3a5b5abe45R71

Reply:
My bad, I missed that one. Thanks!